internal/services/firewall/firewall_resource.go

```diff
@@ -578,6 +629,34 @@ func resourceFirewallDelete(d *pluginsdk.ResourceData, meta interface{}) error {
 	return err
 }
 
+func expandAutoscaleConfiguration(config []interface{}) (*azurefirewalls.AzureFirewallAutoscaleConfiguration, error) {
+	configuration := azurefirewalls.AzureFirewallAutoscaleConfiguration{}
+	emptyConfig := make(map[string]interface{})
+	if len(config) == 1 {
+		emptyConfig = config[0].(map[string]interface{})
+	}
+
+	minPresent := false
+	maxPresent := false
```

Comment:
These are not necessary as for the nested block like `autoconfig`, once it returns the internal object, all the fields are present in the map, due to the implementation detail of the underlying legacy SDK.